Remotion/Data/DomainObjects/ConfigurationLoader/ReflectionBasedConfigurationLoader/LegacyPropertyDefaultValueProvider.cs

```diff
@@ -0,0 +1,64 @@
+using System;
+using System.Collections.Concurrent;
+using System.Linq;
+using System.Runtime.Serialization;
+using Remotion.Data.DomainObjects.Mapping;
+using Remotion.ExtensibleEnums;
+using Remotion.Reflection;
+using Remotion.Utilities;
+
+namespace Remotion.Data.DomainObjects.ConfigurationLoader.ReflectionBasedConfigurationLoader
+{
+  /// <summary>
+  /// Implements the legacy behavior for <see cref="PropertyDefinition.DefaultValue"/>, which sets non-nullable reference-<see cref="Type"/> properties to
+  /// non-<see langword="null"/> values. This affects all <see cref="Array"/>, <see cref="string"/>, and <see cref="ExtensibleEnum{T}"/> properties.
+  /// </summary>
+  /// <threadsafety static="true" instance="true" />
+  public class LegacyPropertyDefaultValueProvider : IPropertyDefaultValueProvider
+  {
+    private static readonly ConcurrentDictionary<Type, IExtensibleEnum> s_uninitializedValues = new();
+    private static readonly object?[] s_extensibleEnumValues = { "Uninitialized" };
+
+    public object? GetDefaultValue (IPropertyInformation propertyInfo, bool isNullable)
+    {
+      if (isNullable)
+        return null;
+
+      var propertyType = propertyInfo.PropertyType;
+      if (propertyType.IsArray)
+      {
+        var elementType = propertyType.GetElementType();
+        Assertion.DebugIsNotNull(elementType, "elementType != null when _propertyType.IsArray");
+        return Array.CreateInstance(elementType, 0);
+      }
+
+      if (propertyType == typeof(string))
+        return string.Empty;
+
+      if (propertyType.IsEnum)
+      {
+        var firstValueOrNull = EnumUtility.GetEnumMetadata(propertyType).OrderedValues.FirstOrDefault();
+        if (firstValueOrNull == null)
+          firstValueOrNull = Activator.CreateInstance(propertyType);
+
+        return firstValueOrNull;
+      }
+
+      if (ExtensibleEnumUtility.IsExtensibleEnumType(propertyType))
+      {
+        var firstValueOrNull = ExtensibleEnumUtility.GetDefinition(propertyType).GetValueInfos().FirstOrDefault()?.Value;
+        return firstValueOrNull ?? s_uninitializedValues.GetOrAdd(propertyType, CreateUninitializedExtensibleEnumValue);
+      }
+
+      return propertyType.IsValueType ? Activator.CreateInstance(propertyType) : null;
+    }
+
+    private IExtensibleEnum CreateUninitializedExtensibleEnumValue (Type extensibleEnumType)
+    {
+      var uninitializedObject = FormatterServices.GetSafeUninitializedObject(extensibleEnumType);
+      var members = FormatterServices.GetSerializableMembers(extensibleEnumType).Where(m=>m.Name.Contains(nameof(IExtensibleEnum.ValueName))).ToArray();
+      FormatterServices.PopulateObjectMembers(uninitializedObject, members, s_extensibleEnumValues);
```

Comment:
I'm not sure about the impelmentation here. 
a) there's no guarantee that ExtensibleEnum is implemented as serializable (don't know if that's requred). Anyway, I would suggest using Type.GetFields instead.
b) There's no guarantee that the fieldname matches. I would suggest we simply go and initalize all string fields on the enum.

Reply:
How do I initialize all string fields if I don't know what they mean? I only know `ValueName` and `DeclarationSpace`, because those are defined on `ExtensibleEnum<T>`.
We might be overthinking this, since this "uninitialized" value only exists so we don't have to throw an exception here, but let the mapping validation handle it.

Reply:
The problem is, we can't make sure we identify the correct fields, it's just a heuristic that could be correct. Since we don't use the enum object for more than a dummy value and have the system fail via validation anyway, we can also just skip the entire exercise and leave the object with null-values in their fields. less code, same result :)

Reply:
Because `ValueName` is not nullable, I thought we cannot leave it at `null`, but now that you mention it - why not?
Do you think the `ConcurrentDictionary` is valuable, or should we get rid of that as well?

Reply:
ad 1) ValueName: exactly, no reason to not leave it null
ad 2) ConDict: Since we don't need to test/detect them, I think we can drop it.